From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: mattprost <matt.prost@ni.com>
Date: Tue, 6 Oct 2020 18:23:35 -0500
Subject: [PATCH] b200: handle overruns during continuous streaming

This allows the b200 devices to recover from overruns that occur during
continuous streaming.

Signed-off-by: mattprost <matt.prost@ni.com>
---
 host/lib/usrp/cores/rx_vita_core_3000.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/host/lib/usrp/cores/rx_vita_core_3000.cpp b/host/lib/usrp/cores/rx_vita_core_3000.cpp
index 4b09f75fd..368ae8e0a 100644
--- a/host/lib/usrp/cores/rx_vita_core_3000.cpp
+++ b/host/lib/usrp/cores/rx_vita_core_3000.cpp
@@ -117,6 +117,9 @@ struct rx_vita_core_3000_impl : rx_vita_core_3000
         cmd_word |= uint32_t((inst_stop) ? 1 : 0) << 28;
         cmd_word |= (inst_samps) ? stream_cmd.num_samps : ((inst_stop) ? 0 : 1);
 
+        _continuous_streaming = stream_cmd.stream_mode
+                                == stream_cmd_t::STREAM_MODE_START_CONTINUOUS;
+
         // issue the stream command
         _iface->poke32(REG_CTRL_CMD, cmd_word);
         const uint64_t ticks =
-- 
2.30.2

